src/coreclr/jit/gentree.h

```diff
@@ -4785,6 +4785,10 @@ class CallArgs
     // Updateable flag, set to 'true' after we've done any required alignment.
     bool m_alignmentDone : 1;
 #endif
+#ifdef SWIFT_SUPPORT
+    // True if we are calling a Swift method that may throw.
+    bool m_hasSwiftErrorHandling : 1;
```

Comment:
Was the other approach costing TP, or why are we going with this?

I would follow the pattern in `AddedWellKnownArg` instead, unless it really turns out to be costly. I think the consistency and correct-by-construction handling is preferable.

Reply:
(But I liked your initial approach. I don't think the iteration of the args for swift calls is a problem.)

Reply:
> Was the other approach costing TP, or why are we going with this?

The cost of that approach was trivial, though I realized we check if a call uses Swift error handling in a few places where the call may be managed, so we'd have to call `GetUnmanagedCallConv` instead of checking `unmgdCallConv` directly to avoid UB. Considering we already use this `AddedWellKnownArg` state handling in a few places, it seemed cleaner to just copy that approach.

> I would follow the pattern in AddedWellKnownArg instead, unless it really turns out to be costly. I think the consistency and correct-by-construction handling is preferable.

Got it, fixed.

Reply:
`AddedWellKnownArg` is used just to improve performance. If we can avoid introducing extra state to maintain and there is no discernable TP cost of it, then I think we should do that.

Reply:
I checked the TP impact locally, and the `AddedWellKnownArg` actually seems to be a bit more costly in the general case (~0.01-0.02% impact versus 0.07%). Even with the additional branch in `HasSwiftErrorHandling` from the call to `GetUnmanagedCallConv`, that seems like the way to go. I'll revert to it.